### DISCUSSION TITLE
ヒューマノイドロボットの設計（ハードウェア）

### DISCUSSION THREAD

**Discussion by Yusuke-Yamasaki-555** [open-rdc/Bipedal_Robot_Hardware#2] [Ideas]
https://github.com/open-rdc/Bipedal_Robot_Hardware/discussions/1#discussioncomment-11580542
以上の要求仕様に沿ってロボットの設計を進めていく。
この議論では、主にハードウェアを扱う。

回路はこちら：
https://github.com/open-rdc/Bipedal_Robot_Hardware/discussions/3

ソフトはこちら：
https://github.com/open-rdc/Bipedal_Robot_Hardware/discussions/4

**Comment by Yusuke-Yamasaki-555** [comment]
モータの選定を進める。
以下の内容を、ここでも同様に行う。
https://github.com/open-rdc/Bipedal_Robot_Hardware/discussions/1#discussioncomment-11578320

---

## 膝関節に最低限必要なトルクの推定
- 静力学のみで考える。
- Pitch軸のみ考慮し、X-Z軸からなる２次元平面を考える。
- 質量は複数の質点で扱い、腰中心、腰膝間リンク（リンク１）中心、膝足首間リンク（リンク２）中心に置く。
- パラメータは、各質点の質量 [kg] 、各リンク長さ [m] 、各関節角度 [deg] からなる。
- プログラムとモデルは以下を用いる。（手書きのモデルと式が含まれた画像と、それをおこしたpythonプログラムが含まれている）
[humanoid_model_torque_check.zip](https://github.com/user-attachments/files/18147239/humanoid_model_torque_check.zip)

---
### 想定１
- 質量
  - 腰：9
  - リンク１：0.5
  - リンク２：0.5
- 角度（右手系。90 [deg] で床に対して垂直）
  - 腰関節：90
  - 膝関節：65
  - 足関節：115
- リンク長さ
  - 腰重心間リンク：0.1
  - リンク１：0.3
  - リンク２：0.3

結果：各関節トルク [Nm] (右手系なので、負の値は時計回りの回転方向を持つことを示す)
- 腰関節：  -5.40620
- 膝関節：-11.50483
- 足関節：   0.62188
---
### 想定２

**Comment by yasuohayashibara** [comment]
以下は各関節ごとの値でしょうか．それとも2つの関節を合わせた値でしょうか．
各関節の値である場合，定格トルク（7.5 N・m）を超えているため，想定している姿勢や重量を見直し，トルクが7.5 N・m以下となるように調整したほうがよいかと思います．
常に加わる力が定格トルクを超えないように設計することをおすすめします．

膝関節：-11.50483
